### PR TITLE
fix(gateway): update user information payload

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]
This will most likely break some other functionality.
Changes were introduced here with the middle name PR that was required for Uganda. Now I wonder what can we do about it...

**Comment by rikukissa** [comment]
Yea, it's been agreed that the `given: string[]` is to be interpreted as 

```
given: ['Mary Jane', 'Mathilda']
                                     ^--- middle name
                  ^--- first name
```

**Review by rikukissa — APPROVED** [review]

File: packages/gateway/src/features/user/root-resolvers.ts [hunk 1]
```diff
@@ -635,7 +635,7 @@ function createOrUpdateUserPayload(user: GQLUserInput): IUserPayload {
     name: user.name.map((name: GQLHumanNameInput) => ({
       use: name.use as string,
       family: name.familyName?.trim() as string,
-      given: (name.firstNames || '')?.trim().split(' ') as string[]
+      given: [name.firstNames || ''] as string[]
```
Comment (Zangetsu101) [inline packages/gateway/src/features/user/root-resolvers.ts:638]:
```suggestion
      given: [name.firstNames?.trim() || ''] as string[]
```

**Review by Zangetsu101 — APPROVED** [review]

File: packages/gateway/src/features/user/root-resolvers.ts [hunk 2]
```diff
@@ -635,7 +635,7 @@ function createOrUpdateUserPayload(user: GQLUserInput): IUserPayload {
     name: user.name.map((name: GQLHumanNameInput) => ({
       use: name.use as string,
       family: name.familyName?.trim() as string,
-      given: (name.firstNames || '')?.trim().split(' ') as string[]
+      given: [name.firstNames?.trim() || ''] as string[]
```
Comment (rukshn) [inline packages/gateway/src/features/user/root-resolvers.ts:638]: Without the split the type is not an array string, but a string. *The type is incorrect*

Comment (Nil20) [reply packages/gateway/src/features/user/root-resolvers.ts:638]: `name.firstNames?.trim() || ''` it is inside an array here.

**Pull request merged**